prediction_market_agent/agents/autogen_general_agent/social_agent.py

```diff
@@ -0,0 +1,200 @@
+from datetime import datetime, timedelta, timezone
+from enum import Enum
+from string import Template
+from typing import Any, Dict, Optional
+
+import autogen
+from autogen import AssistantAgent, UserProxyAgent
+from autogen.cache import Cache
+from loguru import logger
+from prediction_market_agent_tooling.markets.omen.data_models import OmenBet
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.utils import utcnow
+from pydantic import BaseModel
+from web3 import Web3
+
+from prediction_market_agent.agents.autogen_general_agent.prompts import (
+    CRITIC_PROMPT,
+    INFLUENCER_PROMPT,
+)
+from prediction_market_agent.utils import APIKeys
+
+
+# We have an influencer and a critic. Based on https://microsoft.github.io/autogen/docs/notebooks/agentchat_nestedchat/.
+
+
+# Added this class because properties are not automatically dumped when doing class.dict(include=...),
+# hence new class with less fields is the easiest approach.
+class BetInputPrompt(BaseModel):
+    title: str
+    boolean_outcome: bool
+    collateralAmountUSD: float
```

Comment:
Can we make this market-type agnostic? i.e. not specify currency

Reply:
I thought this class `BetInputPrompt` was indeed market-agnostic - the `collateralAmountUSD` is also present on Manifold markets (via Mana > float), so not sure I follow.

Reply:
[Here](https://github.com/gnosis/prediction-market-agent-tooling/blob/a30f7a99b60cf8f69ee91b066a45fa5ba902c82d/prediction_market_agent_tooling/markets/manifold/data_models.py#L123) manifold bets have an amount in `Mana`, not USD. So I guess I just mean either:
- change the naming of the `collateralAmountUSD` field to `collateralAmount`
- or use a BetAmount type here instead

But in general can we make this agent not omen-specific? i.e. not require 
```
from prediction_market_agent_tooling.markets.omen.data_models import OmenBet
from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
    OmenSubgraphHandler,
)
```
so that we can just specify the market type when creating the farcaster agent.

Reply:
Done